bundles/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/internal/handler/TeslaVehicleHandler.java

```diff
@@ -233,11 +233,16 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         TeslaChannelSelector selector = TeslaChannelSelector.getValueSelectorFromChannelID(channelID);
 
         if (command instanceof RefreshType) {
-            if (isAwake()) {
-                // Request the state of all known variables. This is sub-optimal, but the requests get scheduled and
-                // throttled so we are safe not to break the Tesla SLA
-                requestAllData();
+            if (!isAwake()) {
+                logger.debug("Waking vehicle to refresh all data");
+                wakeUp();
             }
+
+            setActive();
+
+            // Request the state of all known variables. This is sub-optimal, but the requests get scheduled and
+            // throttled so we are safe not to break the Tesla SLA
+            requestAllData();
```

Comment:
If I see it right, this call will not have any effect. The method `isAwake()` will still return false at this time and thus no requests are sent out.
But once the wake up was successful and the vehicle status changes to online, the binding will automatically refresh all data.

Reply:
As stated above, when the car is actually awake, but we think it's inactive, but it shouldn't be anymore, this will actually make requests go out.